.github/workflows/test.yml

```
@@ -28,11 +28,13 @@ jobs:
             ${{ runner.os }}-go-cache
       - name: Run gofmt
         run: gofmt -d -e . 2>&1 | tee outfile && test -z "$(cat outfile)" && rm outfile
-      - name: Run golint
+      - name: Run staticcheck
         run: |
-          go install golang.org/x/lint/golint@latest
-          golint -set_exit_status ./...
-          cd _examples && golint -set_exit_status ./... && cd ..
+          go run honnef.co/go/tools/cmd/staticcheck github.com/cucumber/godog
+          go run honnef.co/go/tools/cmd/staticcheck github.com/cucumber/godog/cmd/godog
```

```suggestion
          go run honnef.co/go/tools/cmd/staticcheck@latest github.com/cucumber/godog
          go run honnef.co/go/tools/cmd/staticcheck@latest github.com/cucumber/godog/cmd/godog
```

Let's try `@latest`, I think it should run without modifying local `go.mod` / `go.sum`.

---

We talked about that, but we were thinking it could be problematic if they released a new version of staticcheck with new rules and our build started to fail the next time someone pushed and our CI downloaded the new version.

Does that make sense?

---

Yes, sure, let's use fixed version then
```
go run honnef.co/go/tools/cmd/staticcheck@v0.2.2 github.com/cucumber/godog
```

`go run` works slightly different with and without version suffix.

---

@vearutop are you thinking it's better if we keep `go.mod` / `go.sum` for _runtime_ dependencies, as opposed to for developer tools?

---

Will try running with the fixed version suffix

---

Yes, I think versioning CI tools with `go.mod` is a bit of an abuse, it is not a bad thing, but it grows dependency graph of a module and may potentially affect library consumers as a transitive dependency.

My favorite approach for CI tools is to download pre-built binaries of them.

---

Where would you like the staticcheck binary stored?